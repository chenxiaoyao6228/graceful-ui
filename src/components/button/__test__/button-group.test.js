import { render } from "@testing-library/vue";
import ButtonGroup from "../button-group.vue";
import Button from "../button.vue";

describe("ButtonGroup", () => {
  let ButtonGroupTester = {
    components: { ButtonGroup, Button },
    template: `
        <ButtonGroup data-testid="button-group"><Button icon="left">上一页</Button><Button icon="right" iconPosition="right">下一页</Button></ButtonGroup>`,
  };
  test("match snapshots", () => {
    const { container } = render(ButtonGroupTester, {});
    expect(container).toMatchSnapshot();
  });

  test("validate child", () => {
    const { getByTestId } = render(ButtonGroupTester);
    expect(getByTestId('button-group').childNodes.length).toBe(2);
  });
});
