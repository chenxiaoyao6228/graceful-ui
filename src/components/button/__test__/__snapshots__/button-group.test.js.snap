// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup match snapshots 1`] = `
<div>
  <div
    class="g-button-group"
  >
    <button
      class="g-button g-button-default g-button-size-default"
    >
      <svg
        class="g-icon icon"
      >
        <use
          aria-hidden="true"
          xlink:href="#icon-left"
        />
      </svg>
       
      <!---->
       
      <div
        class="content"
      >
        上一页
      </div>
    </button>
    <button
      class="g-button g-button-default g-button-size-default icon-right"
    >
      <svg
        class="g-icon icon"
      >
        <use
          aria-hidden="true"
          xlink:href="#icon-right"
        />
      </svg>
       
      <!---->
       
      <div
        class="content"
      >
        下一页
      </div>
    </button>
  </div>
</div>
`;
